### PR TITLE
Remove Unity

### PR DESCRIPTION
**Pull request by costleya**
## Description
Removal of unity-specific code from the SDK.

## Motivation and Context
We are transitioning support to .NET Standard 2.0.

## Testing
Built and ran regular build process for daily releases.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ X] My code follows the code style of this project
- [ X] My change requires a change to the documentation
- [ X] I have updated the documentation accordingly
- [ X] I have read the **README** document
- [ ] I have added tests to cover my changes
- [X ] All new and existing tests passed

## License

- [ X] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Core/AWSSDK.Core.Android.csproj [hunk 1]
```diff
@@ -186,10 +186,7 @@
     <Compile Include="Amazon.Util\_pcl\*.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Library\*.cs">
-      <SubType>Code</SubType>
-    </Compile>
-    <Compile Include="Library\UnityAssemblies\*.cs">
+    <Compile Include="destinationDirectoryPath\*.cs">
```
Comment (matteo-prosperi) [inline sdk/src/Core/AWSSDK.Core.Android.csproj]: This looks weird

File: sdk/src/Core/AWSSDK.Core.PCL.csproj [hunk 2]
```diff
@@ -183,10 +183,7 @@
     <Compile Include="Amazon.Util\_pcl\*.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Library\*.cs">
-      <SubType>Code</SubType>
-    </Compile>
-    <Compile Include="Library\UnityAssemblies\*.cs">
+    <Compile Include="destinationDirectoryPath\*.cs">
```
Comment (matteo-prosperi) [inline sdk/src/Core/AWSSDK.Core.PCL.csproj]: This one also

File: docgenerator/SDKDocGeneratorLib/output-files/unity-v3-migration.html [hunk 3]
```diff
@@ -1,101 +0,0 @@
-<html>
```
Comment (matteo-prosperi) [inline docgenerator/SDKDocGeneratorLib/output-files/unity-v3-migration.html]: Ideally you should have a redirect or a placeholder if you remove a doc page.

Comment (costleya) [reply docgenerator/SDKDocGeneratorLib/output-files/unity-v3-migration.html]: Ideally.

I will address all of these in one-go before final release.

**Pull request merged**